### PR TITLE
[SYCl][L0] Buffer for multi-device context is using device allocation now

### PR DESCRIPTION
**Pull request by smaslov-intel**
Signed-off-by: Sergey V Maslov <sergey.v.maslov@intel.com>

### CODE REVIEW

**Comment by smaslov-intel** [comment]
https://github.com/intel/llvm/pull/5966

File: SYCL/Plugin/interop-level-zero-buffer.cpp [hunk 1]
```diff
@@ -90,11 +90,6 @@ int main() {
     Queue.wait();
 
     {
-      char *Ptr = (char *)HostBuffer1;
-      for (int i = 0; i < 10; i++) {
-        assert(Ptr[i] == 'a');
-      }
```
Comment (smaslov-intel) [inline SYCL/Plugin/interop-level-zero-buffer.cpp:96]: @againull : from the spec you crafted:
> A SYCL buffer that is constructed with this interop API uses the Level Zero memory allocation for its full lifetime, and the contents of the Level Zero memory allocation are unspecified for the lifetime of the SYCL buffer. If the application modifies the contents of that Level Zero memory allocation during the lifetime of the SYCL buffer, the behavior is undefined.

**Review by againull — APPROVED** [review]

**Comment by smaslov-intel** [comment]
> FAIL: SYCL :: Basic/context-with-multiple-devices.cpp (114 of 877)
[2022-04-12T00:58:00.712Z] ******************** TEST 'SYCL :: Basic/context-with-multiple-devices.cpp' FAILED ********************
[2022-04-12T00:58:00.712Z] Script:
[2022-04-12T00:58:00.712Z] --
[2022-04-12T00:58:00.712Z] : 'RUN: at line 3';    /netbatch/donb3259726_00/runDir/jenkins-dir/workspace/SYCL_CI/intel/Lin/LLVM_Test_Suite/llvm.obj/bin/clang++      -fsycl -fintelfpga -fsycl-unnamed-lambda /netbatch/donb3259726_00/runDir/jenkins-dir/workspace/SYCL_CI/intel/Lin/LLVM_Test_Suite/llvm-test-suite/SYCL/Basic/context-with-multiple-devices.cpp -o /netbatch/donb3259726_00/runDir/jenkins-dir/workspace/SYCL_CI/intel/Lin/LLVM_Test_Suite/build/SYCL/Basic/Output/context-with-multiple-devices.cpp.tmp2.out
[2022-04-12T00:58:00.712Z] : 'RUN: at line 4';   env CL_CONFIG_CPU_EMULATE_DEVICES=2 /netbatch/donb3259726_00/runDir/jenkins-dir/workspace/SYCL_CI/intel/Lin/LLVM_Test_Suite/build/SYCL/Basic/Output/context-with-multiple-devices.cpp.tmp2.out
[2022-04-12T00:58:00.712Z] --
[2022-04-12T00:58:00.712Z] Exit Code: -11

I cannot reproduce this locally. @againull , @cperkinsintel any idea?

**Comment by cperkinsintel** [comment]
Basic/context-with-multiple-devices.cpp  is a known failure we are trying to track down. I have a PR to disable it ( https://github.com/intel/llvm-test-suite/pull/983 ) however, I strongly suspect that if we disable it, the failure will simply show up somewhere else. Seems to be some sort of infrastructure problem. Not sure.

**Pull request merged**